indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -140,11 +138,16 @@
   // tasks that are complete but not cleaned up yet
   private final RemoteTaskRunnerWorkQueue completeTasks = new RemoteTaskRunnerWorkQueue();
 
-  private final ExecutorService runPendingTasksExec = Executors.newSingleThreadExecutor();
+  private final ExecutorService runPendingTasksExec;
 
   // Workers that have been marked as lazy. these workers are not running any tasks and can be terminated safely by the scaling policy.
   private final ConcurrentMap<String, ZkWorker> lazyWorkers = new ConcurrentHashMap<>();
 
+  // workers which were assigned a task and are yet to acknowledge same.
+  // Map: workerId -> taskId
+  private final ConcurrentMap<String, String> workersWithUnacknowledgedTask = new ConcurrentHashMap<>();
+  // tasks which are being tried to be assigned to a worker
+  private final ConcurrentMap<String, String> tryAssignTasks = new ConcurrentHashMap<>();
```

Comment:
will add the comment, ideally I wanted to have a simple ConcurrentSet which does not exist so used ConcurrentMap instead.